### DISCUSSION TITLE
⚒ The Fixes Release

### DISCUSSION THREAD

**Discussion by TheLimeGlass** [SkriptLang/Skript#5551] [General]
# Skript 2.7.0-beta2
Second beta for Skript 2.7 addressing a handful of bugs.

## ⚠ BETA Release Warning
This is a **BETA** release, which means it is not yet meant for production servers. It may contain bugs or unexpected behaviors in its current state.
We recommend you extensively test this release on a local server before taking the risk of using it on a production server (if you truly wish to).
We do have many tests in place to ensure the integrity of Skript remains stable to some degree.

## ⚠ Compatibility Warning
Due to the major API changes within the 2.7.X updates, some addons may no longer work properly. Please be patient as addon developers work to update their addons. We expect to publish a new release of our [AddonPatcher](https://github.com/SkriptLang/AddonPatcher) in the near future, but be aware that it cannot fix all issues.

## 📢 End of 1.12.2 Support
Yet again we mention that we have dropped support for 1.12 and lower (legacy versions) since Skript **2.7**.
This means that this (and any 2.7.X) version will only work on Minecraft versions 1.13 and higher.
Only critical fixes will be backported to 2.6.X -- Report any if found.

### [Click here to view the full list of commits made since 2.7.0-beta1](https://github.com/SkriptLang/Skript/compare/2.7.0-beta1...master)

Click below to view the entire changelog for this release.
<details>
<summary>
<i>Full changelog</i>
</summary>

### New Features
- Updated to 1.19.4. New entities are added. Aliases automatically generate for the exact name. Human made aliases not ready yet.

### Bug Fixes
- Fixed a bug where attacker could not be used in some damage events.
- Fixed some comparisons that could fail because of the changes in 2.7 to ItemMeta comparison.
- Fixed a bug where the item amount changer wouldn't apply when the type was an itemstack. (Thanks @sovdeeth)
- Fixed a bug that prevented usage of variables that were not a default variable when using the default variables structure (Closes https://github.com/SkriptLang/Skript/issues/5510)
- Fixed a bug where the 'raw string' expression would return the formatted string of a non-literal. (Closes https://github.com/SkriptLang/Skript/issues/5515)
- Gradle will now apply Java 17 to your build when compiling. This means it won't matter what version of Java your machine has, Skript will always be set to Java 17. Ensure you run `gradlew clean build` when building for the first time. This is due to Paper 1.17+ requiring Java 17. Skript still targets Java 8 (MC versions under 1.17 can still run Java 8), so your contributions should build target Java 8.
- Fixed an issue where the existing experience orb experience was not respected when setting the exp value in EffDrop. (Closes https://github.com/SkriptLang/Skript/issues/5490) (Thanks @sovdeeth)
- Fixed a bug that happens when the entity changer would update the inventory of players even if there were no items involved with the change, causing unexpected behaviors. (Closes https://github.com/SkriptLang/Skript/issues/5270)
- The second bug happens when trying to remove an inventory with air items from an entity. (Closes https://github.com/SkriptLang/Skript/issues/5237 and https://github.com/SkriptLang/Skript/issues/5396)

### Skript API (Addon developers)
- Added debugging and JProfiler to the gradle tasks. Read more at https://github.com/SkriptLang/Skript/pull/5299 (Thanks @kiip1)
- Fixed an issue where add-ons would have to update to the newly changed converter package move. Added deprecated methods to support add-ons for the meantime.
- Fixed an issue where Skript couldn't be built using `gradlew build`
</details>

## 📝 Help Us Test
We have an [official Discord community](https://discord.gg/ZPsZAg6ygu) for beta testing Skript's new features and releases.

<hr /><em>This discussion was created from the release <a href='https://github.com/SkriptLang/Skript/releases/tag/2.7.0-beta2'>⚒ The Fixes Release</a>.</em>

**Comment by AdrisLT0** [comment]
please transfer it to bukkit or spigot so aternos could have it :) i would, but i dont want to get copyrighted

**Reply by AyhamAl-Ali** [reply]
See https://github.com/SkriptLang/Skript/issues/3678
